server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -2079,16 +2055,21 @@ private void removeFullyOverlapFiles(TsFileResource newTsFile, Iterator<TsFileRe
   private void removeFullyOverlapFile(TsFileResource tsFileResource,
       Iterator<TsFileResource> iterator
       , boolean isSeq) {
+    logger.info("Removing a covered file {}, closed: {}", tsFileResource, tsFileResource.isClosed());
     if (!tsFileResource.isClosed()) {
-      // also remove the TsFileProcessor if the overlapped file is not closed
-      long timePartition = tsFileResource.getTimePartition();
-      Map<Long, TsFileProcessor> fileProcessorMap = isSeq ? workSequenceTsFileProcessors :
-          workUnsequenceTsFileProcessors;
-      TsFileProcessor tsFileProcessor = fileProcessorMap.get(timePartition);
-      if (tsFileProcessor != null && tsFileProcessor.getTsFileResource() == tsFileResource) {
-        //have to take some time to close the tsFileProcessor
-        tsFileProcessor.syncClose();
-        fileProcessorMap.remove(timePartition);
+      try {
+        // also remove the TsFileProcessor if the overlapped file is not closed
+        long timePartition = tsFileResource.getTimePartition();
+        Map<Long, TsFileProcessor> fileProcessorMap = isSeq ? workSequenceTsFileProcessors :
+            workUnsequenceTsFileProcessors;
+        TsFileProcessor tsFileProcessor = fileProcessorMap.get(timePartition);
+        if (tsFileProcessor != null && tsFileProcessor.getTsFileResource() == tsFileResource) {
+          //have to take some time to close the tsFileProcessor
+          tsFileProcessor.syncClose();
+          fileProcessorMap.remove(timePartition);
+        }
+      } catch (Exception e) {
+        logger.error("Cannot close {}", tsFileResource, e);
```

Comment:
The exception will also only be logged in the upper level, so it is fine to do the logging here. When there is any exception, removing it may not be safe anymore, we'd better leave it be.